warp10/src/main/java/io/warp10/script/functions/TOBIN.java

```diff
@@ -17,57 +17,69 @@
 package io.warp10.script.functions;
 
 import io.warp10.script.NamedWarpScriptFunction;
-import io.warp10.script.WarpScriptStackFunction;
 import io.warp10.script.WarpScriptException;
 import io.warp10.script.WarpScriptStack;
-
-import org.bouncycastle.util.encoders.Hex;
+import io.warp10.script.WarpScriptStackFunction;
 
 import java.nio.charset.StandardCharsets;
+import java.util.BitSet;
 
 /**
- * Encode a String in binary
+ * Encode a String, byte[], Long or BitSet in binary
  */
 public class TOBIN extends NamedWarpScriptFunction implements WarpScriptStackFunction {
-  
-  private static final String[] NIBBLES = new String[] {
+
+  private static final String[] NIBBLES = new String[]{
     "0000", "0001", "0010", "0011",
     "0100", "0101", "0110", "0111",
     "1000", "1001", "1010", "1011",
     "1100", "1101", "1110", "1111",
   };
-  
+
   public TOBIN(String name) {
     super(name);
   }
-  
+
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
     Object o = stack.pop();
-  
-    byte[] data = null;
-    
-    if (o instanceof String) {
-      data = o.toString().getBytes(StandardCharsets.UTF_8);
-    } else if (o instanceof byte[]) {
-      data = (byte[]) o;
-    }
-    
-    if (null == data) {
-      throw new WarpScriptException(getName() + " operates on a String or a byte array.");
-    }
 
     StringBuilder sb = new StringBuilder();
-    
-    for (int i = 0; i < data.length; i++) {
-      int nibble = (data[i] >>> 4) & 0xF;
-      sb.append(NIBBLES[nibble]);
-      nibble = (data[i] & 0xF);
-      sb.append(NIBBLES[nibble]);
+
+    if (o instanceof String || o instanceof byte[]) {
+      byte[] data;
+
+      if (o instanceof String) {
+        data = ((String) o).getBytes(StandardCharsets.UTF_8);
+      } else {
+        data = (byte[]) o;
+      }
+
+      for (byte datum: data) {
+        int nibble = (datum >>> 4) & 0xF;
+        sb.append(NIBBLES[nibble]);
+        nibble = (datum & 0xF);
+        sb.append(NIBBLES[nibble]);
+      }
+
+      stack.push(sb.toString());
+    } else if (o instanceof Long) {
+      sb.append("0000000000000000000000000000000000000000000000000000000000000000");
+
+      sb.append(Long.toBinaryString(((Number) o).longValue()));
+      stack.push(sb.substring(sb.length() - 64));
+    } else if (o instanceof BitSet) {
+      BitSet bitset = (BitSet) o;
+
+      for (int i = 0; i < bitset.size(); i++) {
```

Comment:
Shouldn't we use bitset.length() instead of .size()?

Reply:
You're right, `length()` is better than `size()` in our case.